frontend/__tests__/inquiry/inquiries_table.test.tsx

```diff
@@ -0,0 +1,162 @@
+  },
+  {
+    id: "f919b688-f9c6-49cb-87c0-4dab68a4a04c",
+    text: "How is feedback typically given and received within the team?",
+    created_at: "2024-09-22T09:20:53.734830",
+  },
+]
+
+// Mock dependencies
+jest.mock("@tanstack/react-query")
+
+describe("Inquiries Table", () => {
+  const renderComponent = () => render(<InquiriesTable />)
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it("should display empty table when there's no inquiries.", () => {
+    ;(useQuery as jest.Mock).mockReturnValue({
+      data: { data: [] },
+      isPending: false,
+    })
+
+    renderComponent()
+    expect(screen.getAllByRole("row").length).toBe(1) // just the header row
+  })
+
+  // it("should display skeleton text when data is loading.", async () => {
+  //   (useQuery as jest.Mock).mockReturnValue({
+  //     data: null,
+  //     isPending: true,
+  //   });
+  //   render(<InquiriesTable />);
+  //   screen.debug();
+  //   expect(screen.getByTestId("inquiry-placeholder")).toBeInTheDocument();
+  // });
+
+  it("should display correct number of inquiries in table.", () => {
+    ;(useQuery as jest.Mock).mockReturnValue({
+      data: { data: inquiries },
+      isPending: false,
+    })
+    renderComponent()
+    expect(screen.getAllByRole("row").length).toBe(inquiries.length + 1) // header row + inquiries
+  })
+
+  it("should display correct inquiry text.", () => {
+    ;(useQuery as jest.Mock).mockReturnValue({
+      data: { data: [inquiries[0]] },
+      isPending: false,
+    })
+    renderComponent()
+    expect(screen.getByTestId("inquiry-text")).toHaveTextContent(
+      "How is your work-life balance?",
+    )
+  })
+
+  it("should display correct inquiry created date and time, formatted to the user's timzeone.", () => {
+    ;(useQuery as jest.Mock).mockReturnValue({
+      data: { data: [inquiries[0]] },
+      isPending: false,
+    })
+    // Pretend user's in America/Los_Angeles Timezone
+    jest.spyOn(dayjs.tz, "guess").mockReturnValue("America/Los_Angeles")
+    renderComponent()
+    expect(screen.getByTestId("inquiry-datetime")).toHaveTextContent(
+      "Sep 22, 2024 11:20 AM",
+    )
+  })
+
+  it("should display Invalid Date for inquiries with invalid created date and time.", () => {
+    const inquiry = {
+      id: "92bc71bf-f42c-4b56-b55d-fddaf4633550",
+      text: "How is your work-life balance?",
+      created_at: "",
+    }
+    ;(useQuery as jest.Mock).mockReturnValue({
+      data: { data: [inquiry] },
+      isPending: false,
+    })
+    renderComponent()
+    expect(screen.getByTestId("inquiry-datetime")).toHaveTextContent(
+      "Invalid Date",
+    )
+  })
+
+  it("should display inquiries from newest to oldest.", () => {
+    ;(useQuery as jest.Mock).mockReturnValue({
+      data: { data: inquiries },
+      isPending: false,
+    })
+    renderComponent()
+
+    const rows = screen.getAllByRole("row")
+    expect(rows[1]).toHaveTextContent("How is your work-life balance?")
```

Comment:
so this test has the wrong logic. or rather. i have to think about this.  it would be easier if you looked at on the date of the inquiry rather than on the text.  

Another, much less important note:
There might be a very robust way to do this to with a helper function to convert the date strings into numbers or Date objects that can be compared.  You might also be able to use some of the lessThan and greaterThan logic in jest or javascript.

Reply:
Modified the code to use date comparator:
```
    const dateColumns = screen.getAllByTestId("inquiry-datetime")
    const inquiryDates = dateColumns.map(
      (col) => new Date(col.textContent?.trim() ?? ""),
    )
    for (let i = 1; i < inquiryDates.length; i++) {
      // The inquiries used for this test always have valid created_at datetime value
      /* eslint-disable-next-line */
      expect(inquiryDates[i].getTime()).toBeLessThan(
        inquiryDates[i - 1].getTime(),
      )
    }
```